File: cpp/lab03/lab03.cpp
#include <iostream>
#include <string>

int main()
{
	const std::string format = "P3";
	const int width = 256;
	const int height = 256;
	const int color = 255;

	std::cout << format << std::endl << width << " " << height << std::endl << color << std::endl;

	for(int y = 0; y < height; y++)
	{
		for(int x = 0; x < width; x++)
		{
			if(x < y)
			{
				std::cout << 0 << " " << 0 << " " << 0 << std::endl;
			}
			else
			{
				std::cout << 255 << " " << 255 << " " << 255 << std::endl;
			}
		}
	}

	return 0;
} 

File: javascript/lab03_js/lab03.js
const fs = require('fs');

// Init.
const FORMAT = "P3";
const WIDTH = 256;
const HEIGHT = 256;
const COLOR = 255;

const DrawInformation = `${FORMAT}\n${WIDTH} ${HEIGHT}\n${COLOR}\n`;

let DrawScript = DrawInformation;

// Draw. 
function draw() {
	for(let i = 0; i < HEIGHT; i++) {
		for(let j = 0; j < WIDTH; j++) {
			if(i === j || i === 256 - j) {
				DrawScript += `0 0 0\n`;
			} else {
				DrawScript += `255 255 255\n`;
			}
		}
	}
	return DrawScript;
}

function generate() {
	fs.writeFileSync('./lab03_js/lab03.ppm', draw());
}

generate(); 